x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -70,70 +79,156 @@ export function LayerContextMenu(props: VisualizationLayerWidgetProps<State>) {
+              }),
+            },
+            {
+              id: `${idPrefix}right`,
+              label: i18n.translate('xpack.lens.xyChart.axisSide.right', {
+                defaultMessage: 'Right',
+              }),
+            },
+          ]}
+          idSelected={`${idPrefix}${axisMode}`}
+          onChange={(id) => {
+            const newMode = id.replace(idPrefix, '') as YAxisMode;
+            const newYAxisConfigs = [...(layer.yConfig || [])];
+            const existingIndex = newYAxisConfigs.findIndex(
+              (yAxisConfig) => yAxisConfig.forAccessor === accessor
+            );
+            if (existingIndex !== -1) {
+              newYAxisConfigs[existingIndex].axisMode = newMode;
+            } else {
+              newYAxisConfigs.push({
+                forAccessor: accessor,
+                axisMode: newMode,
+              });
+            }
+            setState(updateLayer(state, { ...layer, yConfig: newYAxisConfigs }, index));
+          }}
+        />
+      </EuiFormRow>
+    </EuiForm>
   );
 }
+
+const tooltipContent = {
+  auto: i18n.translate('xpack.lens.configPanel.color.tooltip.auto', {
+    defaultMessage: 'Lens automatically picks colors for you unless you specify a custom color.',
+  }),
+  custom: i18n.translate('xpack.lens.configPanel.color.tooltip.custom', {
+    defaultMessage: 'Clear the custom color to return to “Auto” mode.',
+  }),
+  disabled: i18n.translate('xpack.lens.configPanel.color.tooltip.disabled', {
+    defaultMessage:
+      'Individual series cannot be custom colored when the layer includes a “Split by.“',
+  }),
+};
+
+const autoMessage = i18n.translate('xpack.lens.configPanel.color.auto', {
+  defaultMessage: 'Auto',
+});
+
+const ColorPicker = ({
+  state,
+  setState,
+  layerId,
+  accessor,
+}: VisualizationDimensionEditorProps<State>) => {
+  const index = state.layers.findIndex((l) => l.layerId === layerId);
+  const layer = state.layers[index];
+  const disabled = !!layer.splitAccessor;
+
+  const [color, setColor] = useState(getSeriesColor(layer, accessor));
+
+  const handleColor: EuiColorPickerProps['onChange'] = (text, output) => {
+    setColor(text);
+
+    if (output.isValid || text === '') {
+      const newYConfigs = [...(layer.yConfig || [])];
+      const existingIndex = newYConfigs.findIndex((yConfig) => yConfig.forAccessor === accessor);
+      if (existingIndex !== -1) {
+        if (text === '') {
+          delete newYConfigs[existingIndex].color;
+        } else {
+          newYConfigs[existingIndex].color = output.hex;
+        }
+      } else {
+        newYConfigs.push({
+          forAccessor: accessor,
+          color: output.hex,
+        });
+      }
+      setState(updateLayer(state, { ...layer, yConfig: newYConfigs }, index));
+    }
+  };
+
+  return (
+    <EuiToolTip
+      position="top"
+      content={
+        disabled ? tooltipContent.disabled : color ? tooltipContent.custom : tooltipContent.auto
+      }
+      delay="long"
```

Comment:
This will help get the width to be correct
```suggestion
      delay="long"
      anchorClassName="eui-displayBlock"
```